ocaml/xapi/context.ml

```diff
@@ -309,11 +309,23 @@ let make ?(http_other_config = []) ?(quiet = false) ?subtask_of ?session_id
             " by task " ^ make_dbg [] "" subtask_of
         )
   ) ;
+  let span_details_from_task_name task_name =
+    match String.split_on_char ':' task_name with
+    | [x; y; uuid] when x = "dispatch" && y = "system.isAlive" ->
```

Comment:
The string literals can be matched directly, no need to use `when` for them

Reply:
As in ["dispatch"; "system.isAlive"; uuid] ? Then we'd need to repeat the string literals, wouldn't it be better to use variables for reuse?

Reply:
Yes, I was think about `| ["dispatch"; "system.isAlive"; uuid] ->`. When using `when` you can match against a let-bound string (and use an arbitrary expression) but your code doesn't do that currently and hence my suggestion. If you want to avoid repeated string literals you have to use `when`.